processor/src/errors.rs

```diff
@@ -16,11 +16,9 @@ use std::error::Error;
 pub enum ExecutionError {
     AdviceKeyNotFound(Word),
     AdviceStackReadFailed(u32),
-    MerkleNodeIndex(Felt, Felt),
+    MerkleNodeIndex { depth: Felt, value: Felt },
     MerkleUpdateInPlace,
-    MerkleSetLookupFailed(MerkleError),
-    MerkleSetNotFound([u8; 32]),
-    MerkleSetUpdateFailed(MerkleError),
+    MerkleStoreLookupFailed(MerkleError),
```

Comment:
I'm thinking we should make the following changes here:

* `MerkleNodeIndex` rename to `InvalidNodeIndex`.
* Add `MerkleStoreUpdateFailed`.

Reply:
Probably was outdated; this was done on the previous commit